bot/extensions/language_cog.py

```diff
@@ -145,6 +147,62 @@ async def remove_trigger_word(self, ctx, old_word):
         else:
             await ctx.send(f"Unable to remove **{old_word}**")
 
+    @hybrid_group(name="puns", help="Commands to manage puns")
+    @has_permissions(administrator=True)
+    async def puns_group(self, ctx):
+        if ctx.invoked_subcommand is None:
+            pun_texts_with_ids = map(lambda pun: '{}.\t{}'.format(
+                pun.id, pun.text), Pun.all())
+
+            embed = Embed(
+                color=self.bot.default_color,
+                title=f"Puns",
+                description="\n".join(pun_texts_with_ids)
+            )
+
+            await ctx.send(embed=embed)
+
+    @puns_group.command(name="add", help="Add a pun", usage="{pun_text}")
+    async def add_pun(self, ctx, pun_text):
+        Pun.create(text=pun_text)
+
+        await ctx.send("Pun added.")
+
+    @puns_group.command(name="remove", help="Remove a pun", usage="{pun_id}")
+    async def remove_pun(self, ctx, pun_id: int):
+        pun = Pun.get_by(id=pun_id)
+
+        if pun:
+            await ctx.send("Pun removed.")
+        else:
+            await ctx.send("Pun with id {pun.id} does not exist.")
+
+    @puns_group.command(name="add-word", help="Add a pun word to a pun")
+    async def add_pun_word(self, ctx, pun_id: int, pun_word, emo):
+        pun = Pun.get_by(id=pun_id)
+
+        if pun:
+            if pun_word in map(lambda pun_word: pun_word.word, pun.pun_words):
```

Comment:
This could be a function called `pun.has_word(pun_word)`. You could also do with a sql query instead which would be faster `pun_words.filter(PunWord.word == word).count() > 0`. You might have to change the relationship to lazy dynamic loading. `pun_words = relationship("PunWord", lazy='dynamic', cascade="all, delete-orphan")`

```python
class Pun(Model):
    ...
    pun_words = relationship("PunWord", lazy='dynamic', cascade="all, delete-orphan")
    ...

   def has_word(self, word):
       return self.pun_words.filter(PunWord.word == word).count() > 0
 ```